lib/SILOptimizer/Transforms/CopyForwarding.cpp

```diff
@@ -1833,6 +1915,9 @@ bool TempRValueOptPass::tryOptimizeCopyIntoTemp(CopyAddrInst *copyInst) {
       use->set(copyInst->getSrc());
       break;
     }
+    // ASSUMPTION: no operations that may be handled by this default clause can
+    // destroy tempObj. This includes operations that load the value from memory
```

Comment:
This is a scary assumption. I don't have a concrete suggestion off the top of my head, but we should see if we can assert this perhaps. I am fine with that being in a follow on commit if you want to do it.

Reply:
It was asserted in the previous commit. The assertion was removed because the default clause was considered "cleaner". I added this comment as documentation for the previous commit and don't want to revisit it in this PR.